packages/graphile-build-pg/src/plugins/PgBackwardRelationPlugin.ts

```diff
@@ -131,27 +132,49 @@ export default (function PgBackwardRelationPlugin(
           const isUnique = !!table.constraints.find(
             c =>
               (c.type === "p" || c.type === "u") &&
-              c.keyAttributeNums.length === keys.length &&
+              c.keyAttributeNums.length <= keys.length &&
               c.keyAttributeNums.every((n, i) => keys[i].num === n)
           );
-
           const isDeprecated = isUnique && legacyRelationMode === DEPRECATED;
 
+          const primaryKeyConstraint = table.primaryKeyConstraint;
+          const primaryKeys =
+            primaryKeyConstraint && primaryKeyConstraint.keyAttributes;
+          const uncoveredPrimaryKeys =
+            primaryKeys && keys.every(attr => primaryKeys.includes(attr))
+              ? primaryKeys.filter(attr => !keys.includes(attr))
+              : [];
+          const parameterKeys = uncoveredPrimaryKeys.map(key => ({
+            ...key,
+            sqlIdentifier: sql.identifier(key.name),
+            paramName: inflection.column(key), // inflection.argument(key.name, i)
+          }));
+
           const singleRelationFieldName = isUnique
             ? inflection.singleRelationByKeysBackwards(
                 keys,
                 table,
                 foreignTable,
                 constraint
               )
+            : uncoveredPrimaryKeys.length
+            ? inflection.rowByRelationBackwardsAndUniqueKeys(
+                uncoveredPrimaryKeys,
+                table,
+                foreignTable,
+                constraint,
+                primaryKeyConstraint as PgConstraint // not void
+              )
             : null;
 
-          const primaryKeyConstraint = table.primaryKeyConstraint;
-          const primaryKeys =
-            primaryKeyConstraint && primaryKeyConstraint.keyAttributes;
-
           const shouldAddSingleRelation =
-            isUnique && legacyRelationMode !== ONLY;
+            (isUnique && legacyRelationMode !== ONLY) ||
+            (!isUnique &&
+              !!uncoveredPrimaryKeys.length &&
+              primaryKeyConstraint &&
+              !omit(table, "single") &&
+              !omit(primaryKeyConstraint, "single") &&
+              !omit(constraint, "single"));
```

Comment:
This is a lot of potential `omit`s. I'd like to really tighten up how `omit` works in V5, I think in this case it should be something like `!omit(constraint, "backward")`